src/voting/CIP-voting.md

```diff
@@ -0,0 +1,168 @@
+# `Governance API`
+
+## **Abstract**
+
+This document describe the interface between webpage / web-based stack and cardano wallets. This specificies that API of the javascript object that need to be injected into the web applications in order to support all the Governance features.
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+They enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+## **Namespace**
+
+### **cardano.{walletName}.governance.enable(): Promise\<API>**
+The `cardano.{walletName}.governance.enable()` method is used to enable the governance API. It should request permission from the wallet to enable the API. If permission is granted, the rest of the API will be available. 
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  voting_key: cbor<vkey>,
+  weight: number
+}
+
+```
+
+`voting_key`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
+
+```
+
+`Purpose`: Defines the purpose of the delegations. This is used to limit the scope of the delegations.  For example, a purpose might be a subset of Catalyst proposals, a council election, or even some private purpose (agreed by convention).
+
+
+# **`API`**
+
+## **api.getVotingKey**(address_index: number = 0, account: number = 0, role: number = 0): Promise\<Bip32PublicKey>
+
+Should derive and return the wallets voting public key
+
+### **Returns**
+`cbor\<vkey>` - cbor serialized 64 bytes (x, y) Ed25519 public key  
+
+There should be only a single single voting key per wallet / mnemonic. 
+
+The **`voting_key`** should be derived from the following path. 
+
+```
+m / 1694' / 1815' / account' / role' / address_index'
+```
+
+`1694` (year Voltaire was born) Sets a dedicated `purpose` in the derivation path for the voting profile.  
```

Comment:
Some guidelines for implementation when keys are managed by a hardware device would be great.

Do we expect hardware wallet APIs to add support for this, or should hw-device-connected wallets derive new in-memory keys/mnemonics for voting?

Reply:
No, the actual point of this and `cip-36` is to avoid new mnemonics or in-memory keys being generated. The same mnemonic should be used and a new derivation path be used.

Reply:
Problem is that voting app expects to get a private key, while HW wallets do not export private keys by design. Existing implementations return a key pair from their variant of  `api.getVotingKey`